### PR TITLE
[huggingface_pytorch, huggingface_tensorflow][build] Huggingface inference DLC

### PR DESCRIPTION
**Pull request by philschmid**
*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [x] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

This PR introduces new Hugging Face Deep Learning Container for Inference. It contains CPU and GPU images for PyTorch and TensorFlow. I also tried to adjust the `buildspec.yaml`. 

Looking forward on your feedback. 

*Tests run:*

-

*DLC image/dockerfile:*

The Hugging Face DLCs

*Additional context:*

`docker build` is not yet possible, since the `sagemaker_huggingface_inference_toolkit` is not released. This PR but is ready for review. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: huggingface/pytorch/buildspec.yml [hunk 1]
```diff
@@ -29,3 +44,36 @@ images:
...
+  BuildHuggingFacePytorchCPUPTInferencePy3DockerImage:
+    <<: *INFERENCE_REPOSITORY
+    build: &HUGGINGFACE_PYTORCH_CPU_INFERENCE_PY3 false
+    image_size_baseline: 4899
+    device_type: &DEVICE_TYPE cpu
+    python_version: &DOCKER_PYTHON_VERSION py3
+    tag_python_version: &TAG_PYTHON_VERSION py36
+    os_version: &OS_VERSION ubuntu18.04
+    transformers_version: &TRANSFORMERS_VERSION 4.6.0
+    inference_toolkit_version: &INFERENCE_TOOLKIT_VERSION 1.0.0
```
Comment (philschmid) [inline huggingface/pytorch/buildspec.yml:56]: Not sure if we should this in the `dockerfile` or in the `buildpsec.yaml`

File: huggingface/build_artifacts/inference/config.properties [hunk 2]
```diff
@@ -0,0 +1,5 @@
+vmargs=-XX:+UseContainerSupport -XX:InitialRAMPercentage=8.0 -XX:MaxRAMPercentage=10.0 -XX:-UseLargePages -XX:+UseG1GC -XX:+ExitOnOutOfMemoryError
```
Comment (vdantu) [inline huggingface/build_artifacts/inference/config.properties:1]: -XX:-UseContainerSupport? Did we test this configuration?

File: huggingface/pytorch/buildspec.yml (hunk 1 above)
Comment (vdantu) [reply huggingface/pytorch/buildspec.yml:56]: Good point. Question of usability of Dockerfiles vs code manageability as this version number is used in all the containers. Would recommend leaving it here for now and moving it into Dockerfiles if need arises.

File: huggingface/build_artifacts/inference/config.properties (hunk 2 above)
Comment (vdantu) [reply huggingface/build_artifacts/inference/config.properties:1]: Could we address this comment?

**Review by vdantu — CHANGES_REQUESTED** [review]

File: huggingface/build_artifacts/inference/config.properties (hunk 2 above)
Comment (philschmid) [reply huggingface/build_artifacts/inference/config.properties:1]: Yes, I have done all the current testing with this configuration so far.

**Comment by philschmid** [comment]
@saimidu can you give it a look at why the building fails?

File: huggingface/build_artifacts/inference/config.properties (hunk 2 above)
Comment (philschmid) [reply huggingface/build_artifacts/inference/config.properties:1]: can you add the suggestion?

Comment (vdantu) [reply huggingface/build_artifacts/inference/config.properties:1]: Replace the option `-XX:+UseContainerSupport` above with `-XX:-UseContainerSupport` .. `+` changes to `-`

File: mxnet/inference/docker/1.5.1/py3/Dockerfile.eia [hunk 3]
```diff
@@ -9,6 +9,7 @@ ARG MMS_VERSION=1.1.2
 ARG PYTHON=python3
 ARG PYTHON_VERSION=3.6.10
 ARG HEALTH_CHECK_VERSION=1.7.0
+ARG OPENSSL_VERSION=1.1.1k
```
Comment (vdantu) [inline mxnet/inference/docker/1.5.1/py3/Dockerfile.eia:12]: Why are we modifying MXNet containers as a part of this PR? If this is unintentional, could we revert this back?

File: mxnet/inference/docker/1.6.0/py2/Dockerfile.cpu [hunk 4]
```diff
@@ -15,8 +15,7 @@ ARG MX_URL=https://aws-mxnet-pypi.s3-us-west-2.amazonaws.com/1.6.0/aws_mxnet_mkl
 ARG PYTHON=python
 ARG PYTHON_PIP=python-pip
 ARG PIP=pip
-
-ARG OPENSSL_VERSION=1.1.1g
+ARG OPENSSL_VERSION=1.1.1k
```
Comment (vdantu) [inline mxnet/inference/docker/1.6.0/py2/Dockerfile.cpu:18]: Same here. Could we revert this change. If this was intentional, we could  pull it in as a separate PR.

File: mxnet/buildspec-neuron.yml [hunk 5]
```diff
@@ -1,8 +1,9 @@
   account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
   region: &REGION <set-$REGION-in-environment>
   framework: &FRAMEWORK mxnet
-  version: &VERSION 1.5.1
-  os_version: &OS_VERSION ubuntu16.04
+  version: &VERSION 1.8.0
```
Comment (vdantu) [inline mxnet/buildspec-neuron.yml:4]: Is this change required for HF containers?

File: mxnet/inference/docker/1.8/py3/Dockerfile.neuron [hunk 6]
```diff
@@ -0,0 +1,145 @@
+FROM ubuntu:18.04
```
Comment (vdantu) [inline mxnet/inference/docker/1.8/py3/Dockerfile.neuron:1]: Is this change required for HF containers?

**Review by vdantu — CHANGES_REQUESTED** [review]
The PR seems to have unintentional non-HF changing. We might need to rebase the PR.
